### PR TITLE
Add observations parser

### PR DESCRIPTION
**Pull request by mgrover1**
This is a parser meant to work with the amwg diagnostics package observational datasets

### CODE REVIEW

File: ecgtools/parsers/observations.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import pathlib
+import traceback
+import warnings
```
Comment (andersy005) [inline ecgtools/parsers/observations.py:3]:
```suggestion
```

File: ecgtools/parsers/observations.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                units = da.units
+                long_name = da.long_name
+                info = {
+                    'source': source,
+                    'temporal': temporal,
+                    'time_period': time_period,
+                    'variable': var,
+                    'units': units,
+                    'long_name': long_name,
+                    'path': str(path),
```
Comment (andersy005) [inline ecgtools/parsers/observations.py:40]:
```suggestion
                    'path': str(file),
```

**Comment by andersy005** [comment]
@mgrover1, could you add one or two unit tests for this when you get a moment?

**Comment by mgrover1** [comment]
@andersy005 I added a couple tests here - feel free to take a look when you get a chance

File: tests/test_obs.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+from ecgtools.parsers.observations import parse_amwg_obs
+
+df = pd.DataFrame()
+
+
+@pytest.mark.parametrize(
+    'file_path',
+    [
+        '/glade/p/cesm/amwg/amwg_diagnostics/obs_data/AIRS_01_climo.nc',
+        '/glade/p/cesm/amwg/amwg_diagnostics/obs_data/MODIS_ANN_climo.nc',
```
Comment (andersy005) [inline tests/test_obs.py:13]: Could we add tiny samples (1-2 MB) of these files to the `sample_data` directory within this repository so as to enable running tests without GLADE access dependency?

File: tests/test_obs.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+@pytest.mark.parametrize(
+    'file_directory',
+    [sample_data_dir / 'cesm_obs'],
+)
+def test_obs_builder(file_directory):
+    b = Builder(file_directory)
+    b.build(parse_amwg_obs)
+    print(file_directory)
+    print(b.df)
```
Comment (andersy005) [inline tests/test_obs.py:36]:
```suggestion
```

**Review by andersy005 — APPROVED** [review]

**Pull request merged**